spec/compiler/completeness_spec.cr

```diff
@@ -32,6 +32,58 @@ describe Mare::Compiler::Completeness do
     end
   end
 
+  it "complains when field write is in while loop in constructor" do
+    source = Mare::Source.new_example <<-SOURCE
+    :class Data
+      :prop w U64
+      :new
+        while True (
+          @w = 0
+        )
+    SOURCE
+
+    expected = <<-MSG
+    This constructor doesn't initialize all of its fields:
+    from (example):3:
+      :new
+       ^~~
+
+    - this field didn't get initialized:
+      from (example):2:
+      :prop w U64
+            ^
+    MSG
+
+    expect_raises Mare::Error, expected do
+      Mare::Compiler.compile([source], :completeness)
+    end
+  end
+  it "complains when field write is away in constructor" do
```

Comment:
Small quibble, but an extra empty line should be between each pair of test cases.

```suggestion

  it "complains when field write is away in constructor" do
```